core_lib/src/util/pointerevent.h

```diff
@@ -21,19 +21,20 @@ class PointerEvent
         Unmapped
     };
 
-    PointerEvent(QMouseEvent* event);
-    PointerEvent(QTabletEvent* event);
+    PointerEvent(QMouseEvent* event, const QPointF& canvasPos);
+    PointerEvent(QTabletEvent* event, const QPointF& canvasPos);
     ~PointerEvent();
 
     /**
-     * Returns QPoint of the device */
-    QPoint pos() const;
+     * Returns the QPointF of the device, in canvas coordinates
+     */
+    QPointF canvasPos() const;
 
     /**
-     * Returns the QPointF of the device
+     * Returns the QPointF of the device, in viewport coordinates
      * Returns pos() if used on mouse event
      */
-    QPointF posF() const;
+    QPointF viewportPos() const;
```

Comment:
```suggestion
    const QPointF& viewportPos() const;
```

Reply:
Disregard this, I forgot that we're creating an empty QPointF at the bottom, which wouldn't work with a temporary reference.